src/shared/rpc/methods/sign-psbt.ts

```diff
@@ -12,9 +11,11 @@ import {
   validateRpcParams,
 } from './validation.utils';
 
+const sigHashTypes = [0x01, 0x02, 0x03, 0x80, 0x81, 0x82, 0x83];
+
 const rpcSignPsbtParamsSchema = yup.object().shape({
   account: accountSchema,
-  allowedSighash: yup.array().of(yup.mixed().oneOf(Object.values(btc.SignatureHash))),
+  allowedSighash: yup.array().of(yup.mixed().oneOf(Object.values(sigHashTypes))),
```

Comment:
Not understanding how this will work, bc the sign function you pointed to in the telegram channel calls `signIdx` that types the enum?

Reply:
I suppose the typing doesn't really matter?

Reply:
It's possible it needs to change, right? I also thought that you pass an array of the enum values, but maybe not.

I think the typing of `signIdx` is wrong, and not sure why it isn't a type error

<img width="1069" alt="image" src="https://github.com/hirosystems/wallet/assets/1618764/0052c403-655a-489a-925e-e60d45818afb">

and indeed no type error in this code

```ts
enum TestEnum {
  One, Two, Three
}

function test(dir: TestEnum[]) {}

function testWrapper(arg: number[]) {
  return test(arg); // no type error even though `number[]`
}
```

Auditing library there's this line
```ts
    if (!allowedSighash.includes(sighash)) {
```

Where we describe `allowedSighash` and `sighash` comes from input. If this value returns any of the `0x81` values, then indeed we do need to update our code.

Reply:
Just ran the `btc-signer` tests and they use the other value:

```
{ sighash: 130 }
{ sighash: 130 }
{ sighash: 130 }
{ sighash: 131 }
{ sighash: 131 }
{ sighash: 131 }
{ sighash: 129 }
{ sighash: 129 }
{ sighash: 129 }
{ sighash: 1 }
{ sighash: 1 }
{ sighash: 1 }
```

Reply:
Yep, I do know that `allowedSighash` and the `sighash` on the input are diff. I've been referencing that for awhile. The allowed array means it is allowed, but not if the sighash on the input isn't included. My confusion is, why is his enum so limited then? Seems like an error that needs to be corrected in the lib, or we are missing something about why the enum was constructed that way?